src/main/java/uk/gov/hmcts/reform/ucmc/request/RequestData.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.hmcts.reform.ucmc.request;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.servlet.http.HttpServletRequest;
+
+@Service
+public class RequestData {
```

Comment:
just a note, fpl encountered some issues when using async exection and this. Essentially spring cleans up the request as soon as the controller method is finished so it'll not be there when the async method needs it